source/slang/source-loc.h

```diff
@@ -82,17 +113,21 @@ class SourceFile : public RefObject
     /// The actual contents of the file.
     UnownedStringSlice content;
 
-    // The range of source locations that the span covers
-    SourceRange sourceRange;
-
+    protected:
     // In order to speed up lookup of line number information,
     // we will cache the starting offset of each line break in
     // the input file:
-    List<UInt> lineBreakOffsets;
+    List<uint32_t> m_lineBreakOffsets;
```

Comment:
This change sets our maximum input file size to 4GB instead of "whatever we can fit in memory." I suppose it isn't a big deal because Slang is likely to choke on any file that big anyway, but I just want to make sure we are aware when we are changing design decisions that might have been made on purpose.

Also note that the memory usage for this array shouldn't be something we work very hard to minimize, because (1) it is only used when reporting errors, and I think the compiler is justified in optimizing performance for the non-error case, and (2) the number of lines in a file is typically way less than the number of bytes in the file, so no matter what this array is unlikely to add up to nearly the amount of storage for the bytes of the file itself.

Reply:
Understood. On the other hand this structure may be hit *much* more, than just the error scenario - during serialization of debug information.